cmd/cmdutils.go

```diff
@@ -170,3 +173,51 @@ func printUnmountedStorage(client *occlient.Client, applicationName string) {
 	}
 	fmt.Println("")
 }
+
+// GetAppName returns application name from the provided flag or if flag is not provided, it will return current application name
+func getAppName(client *occlient.Client) string {
+	// applicationFlag is `--application` flag
+	if applicationFlag != "" {
+
+		_, err := application.Exists(client, applicationFlag)
+		if err != nil {
+			checkError(err, "")
+		}
+		return applicationFlag
+	}
+	applicationName, err := application.GetCurrent(client.Namespace)
+	checkError(err, "unable to get current application")
+
+	return applicationName
+}
+
+// getAndSetNamespace checks whether project flag is provided,
+// if provided, it validates the name and sets it as namespace for further operations
+// if not provided, it fetches current namespace and sets it as namespace for further operations
+// getAndSetNamespace also return the project name
+func getAndSetNamespace(client *occlient.Client) string {
+	// projectFlag is `--project` flag
+	if projectFlag != "" {
+		_, err := project.Exists(client, projectFlag)
+		if err != nil {
+			checkError(err, "")
+		}
+		client.Namespace = projectFlag
```

Comment:
Any reason why we aren't using SetCurrentProject and only doing a switch in-memory?
Is it to avoid overhead involved in it? But what happens if for example https://github.com/redhat-developer/odo/pull/683/files#diff-cfbbcc0af92db2bb72c86987124fe72cR61 fails? would it not change namespace even for a failed create(although only in-memory)?

Reply:
To avoid overhead, we are using `client.Namespace` everywhere instead of confusing with new variable `projectName`

I didn't get your second qs, can you please elaborate ?

> But what happens if for example https://github.com/redhat-developer/odo/pull/683/files#diff-cfbbcc0af92db2bb72c86987124fe72cR61 fails? would it not change namespace even for a failed create(although only in-memory)?.

what is in memory state and the reality state ?

Reply:
See for example, if the operation in https://github.com/redhat-developer/odo/pull/683/files#diff-cfbbcc0af92db2bb72c86987124fe72cR61 fails, by then you have set client.Namespace without changing it in config file so, if the operation fails, no change in config file but client.Namespace which is in use has changed so aren't they different was my question?

But it seems, each command is using a new instance of client so this namespace setting would have no impact on them it'll simply be overwritten right? with whatever exists in config...

Ok sounds good...

Reply:
awesome :)